fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/BlobContainerManager.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.blob;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.config.FHIRConfiguration;
+import com.ibm.fhir.config.FHIRRequestContext;
+import com.ibm.fhir.config.PropertyGroup;
+import com.ibm.fhir.core.lifecycle.EventCallback;
+import com.ibm.fhir.core.lifecycle.EventManager;
+
+/**
+ * Singleton to abstract and manage Azure Blob containers.
+ * Each tenant/datasource gets its own container.
+ * 
+ * TODO: investigate if BlobContainerClient should be long-lived and shared
```

Comment:
i'm assuming this will be done as part of #3136

Reply:
Yes, once we get things deployed and can run some larger scale tests we'll hopefully get to see how things behave.